feature_engine/missing_data_imputers.py

```diff
@@ -285,10 +293,16 @@ def fit(self, X, y=None):
         X = _is_dataframe(X)
 
         # find or check for numerical variables
-        self.variables = _find_numerical_variables(X, self.variables)
+        if self.imputer_dict:
+            self.variables = _find_numerical_variables(X, self.imputer_dict.keys())
+        else:
+            self.variables = _find_numerical_variables(X, self.variables)
 
         # create the imputer dictionary
-        self.imputer_dict_ = {var: self.arbitrary_number for var in self.variables}
+        if self.imputer_dict:
+            self.imputer_dict_ = {var: self.imputer_dict[var] for var in self.variables}
```

Comment:
would it not work here if we just do: `self.imputer_dict_ = self.imputer_dict`, so that we avoid the loop?

Because in any case, lines 296-299 will check that the variables are numerical. So if that passes, then I think we are safe to use the dictionary?

Reply:
indeed, good point, I've applied this in next commit.